jetstream_pt/cache_manager.py

```diff
@@ -663,18 +664,21 @@ def __init__(
       self,
       cache_k: torch.Tensor,  # previous cache
       cache_v: torch.Tensor,  # previous cache
+      page_attention_manager: PageAttentionManager,
       page_token_indices: torch.Tensor,  # page and token indices for the cache
       sharding,
       env=None,
   ):
     super().__init__()
     self.cache_k = cache_k
     self.cache_v = cache_v
+    self.page_attention_manager = page_attention_manager
     self.page_token_indices = page_token_indices
     self.sharding = sharding
     self.env = env
+    self.stacked = False
 
-  def update(self, key, value):
+  def update(self, key, value, layer_id=0):
```

Comment:
can pass layer_id from the constructor:
Each layer will get a different cache_manager instance anyways.

Reply:
Good pint! Below are code to call the update in layer.py
`
 orig_keys, orig_values = cache.update(xk, xv, self.layer_id)
`


For page attention, the update didn't use layer_id (it's just a placer holder). I will refactor it later after this PR to isolate the potential for lazy update and normal update.